packages/jellyfish-transaction/src/script/dftx/dftx_loans.ts

```diff
@@ -0,0 +1,33 @@
+import { BufferComposer, ComposableBuffer } from '../../buffer/buffer_composer'
+import BigNumber from 'bignumber.js'
+
+// Disabling no-return-assign makes the code cleaner with the setter and getter */
+/* eslint-disable no-return-assign */
+
+/**
+ * CreateLoanScheme DeFi Transaction
+ */
+export interface CreateLoanScheme {
+  ratio: number // -----------------------| 4 bytes unsigned
+  rate: BigNumber // ---------------------| 8 bytes unsigned
+  identifier: string // ------------------| c = VarUInt{1-9 bytes}, + c bytes UTF encoded string
+  update: BigNumber // -------------------| 8 bytes unsigned integer, activation block height. 0 for createLoanScheme, > 0 for updateLoanScheme
+}
+
+/**
+ * Composable CreateLoanScheme, C stands for Composable.
+ * Immutable by design, bi-directional fromBuffer, toBuffer deep composer.
+ */
+export class CCreateLoanScheme extends ComposableBuffer<CreateLoanScheme> {
+  static OP_CODE = 0x4c // 'L'
+  static OP_NAME = 'OP_DEFI_TX_CREATE_LOAN_SCHEME'
+
+  composers (cls: CreateLoanScheme): BufferComposer[] {
+    return [
+      ComposableBuffer.uInt32(() => cls.ratio, v => cls.ratio = v),
+      ComposableBuffer.satoshiAsBigNumber(() => cls.rate, v => cls.rate = v),
+      ComposableBuffer.varUIntUtf8BE(() => cls.identifier, v => cls.identifier = v),
```

Comment:
@surangap are strings encoded in `utf8` on c++ by default? I never looked at this close enough.

Reply:
we use std::string in c++ side. basically, it is a collection of `char` elements. so it supports holding utf-8.
there's going to be a `std::u8string` type in c++20 which is basically `std::basic_string<char8_t>`, not sure what are the implications to `std::string` are, but that is c++20, we are good for now.

Reply:
Ah thanks, that make sense now